src/categories/models.ts

```diff
@@ -0,0 +1,58 @@
+import { db } from '../config'
+import { ModifyCategoryRequest } from './objects'
+
+const TABLE_NAME = 'categories'
+
+const CATEGORY_FIELDS = [
+  'id',
+  'name',
+  'type',
+  'extra',
+  'description',
+  'pictureUrl',
+  'color'
+]
+
+export interface CategoryModel {
+  id?: number
+  name: string
+  type: string
```

Comment:
This `type` field has a weak data type. There are two approaches to solve this:
* Use typescript enums, https://medium.com/@katbusch/typescript-enums-explained-e5f9a101afc9
* Add string constants that would be need to be used on all the project, instead of comparing to plain strings.

I'm in favour of the second one, but we need the *string* enums version. on the post there are examples of that.

Reply:
I followed the given article and, according to its top suggestion, defined a new string literal type:

```typescript
export type categoryType = 'TO_USER' | 'TO_TWO_USERS' | 'ONLY_EXTRA';
```

Everything has been changed accordingly.

Fixed in 1a92f9a